Content.Server/GameObjects/Components/Kitchen/ReagentGrinderComponent.cs

```diff
@@ -0,0 +1,367 @@
+                    //Remember the beaker can be null!
+                    _heldBeaker?.Solution.Contents.ToArray()
+                ));
+                _dirty = false;
+            }
+        }
+
+        private void EjectSolid(EntityUid entityID)
+        {
+            if (_busy)
+                return;
+
+            if (_entityManager.EntityExists(entityID))
+            {
+                var entity = _entityManager.GetEntity(entityID);
+                _chamber.Remove(entity);
+
+                //Give the ejected entity a tiny bit of offset so each one is apparent in case of a big stack,
+                //but (hopefully) not enough to clip it through a solid (wall).
+                const float ejectOffset = 0.4f;
+                entity.Transform.LocalPosition += (_random.NextFloat() * ejectOffset, _random.NextFloat() * ejectOffset);
+            }
+            _dirty = true;
+        }
+
+        /// <summary>
+        /// If this component contains an entity with a <see cref="SolutionContainerComponent"/>, eject it.
+        /// Tries to eject into user's hands first, then ejects onto dispenser if both hands are full.
+        /// </summary>
+        private void EjectBeaker(IEntity user)
+        {
+            if (!HasBeaker || _busy)
+                return;
+
+            //Eject the beaker into the hands of the user.
+            _beakerContainer.Remove(_beakerContainer.ContainedEntity);
+
+            //UpdateUserInterface();
+
+            if (!user.TryGetComponent<HandsComponent>(out var hands) || !_heldBeaker!.Owner.TryGetComponent<ItemComponent>(out var item))
+                return;
+            if (hands.CanPutInHand(item))
+                hands.PutInHand(item);
+
+            _heldBeaker = null;
+            _dirty = true;
+            SetAppearance(ReagentGrinderVisualState.NoBeaker);
+        }
+
+        void IActivate.Activate(ActivateEventArgs eventArgs)
+        {
+            if (!eventArgs.User.TryGetComponent(out IActorComponent? actor) || !Powered)
+            {
+                return;
+            }
+
+            _dirty = true;
+            UserInterface?.Toggle(actor.playerSession);
+        }
+
+        public async Task<bool> InteractUsing(InteractUsingEventArgs eventArgs)
+        {
+            if (!Powered)
+                return false;
+
+            if (!eventArgs.User.TryGetComponent(out IHandsComponent? hands))
+            {
+                Owner.PopupMessage(eventArgs.User, Loc.GetString("You have no hands."));
+                return true;
+            }
+
+            
+            var heldEnt = eventArgs.Using;
+
+            //First, check if user is trying to insert a beaker.
+            //No promise it will be a beaker right now, but whatever.
+            //Maybe this should whitelist "beaker" in the prototype id of heldEnt?
+            if(heldEnt!.TryGetComponent(out SolutionContainerComponent? beaker) && heldEnt!.Prototype!.ID.ToLower().Contains("beaker"))
```

Comment:
Yeah this is bad, use SolutionContainerCaps

Reply:
fair point good catch